mne/io/snirf/_snirf.py

```diff
@@ -0,0 +1,202 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+import re as re
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ...annotations import Annotations
+from ...utils import logger, verbose, fill_doc, warn, requires_h5py
+
+
+@requires_h5py
+@fill_doc
+def read_raw_snirf(fname, preload=False, verbose=None):
+    """Reader for a SNIRF fNIRS recording.
+
+    .. note:: This reader supports the .snirf file type only,
+              not the .jnirs version.
+              Currently only support for continuous wave SNIRF data is
+              provided.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the SNIRF data file.
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawSNIRF
+        A Raw object containing SNIRF data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawSNIRF(fname, preload, verbose)
+
+
+def _open(fname):
+    return open(fname, 'r', encoding='latin-1')
+
+
+@requires_h5py
+@fill_doc
+class RawSNIRF(BaseRaw):
+    """Raw object from a SNIRF fNIRS file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the SNIRF data file.
+    %(preload)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None):
+        from ...externals.pymatreader.utils import _import_h5py
+        h5py = _import_h5py()
```

Comment:
Ahh, if you're already using this (which probably does what I propose adding internally to MNE above) then you shouldn't have `requires_h5py` above

Reply:
Ok I think I have handled this as you suggest now. But its probably worth double checking I interpreted your suggestions correctly.